app/controllers/dinners_controller.rb

```diff
@@ -6,16 +6,21 @@ def new
   def create
     @dinner = Dinner.new(dinner_params)
     if @dinner.save
-      # TODO -- Flash a success
+      flash[:notice] = "Copy your link and send to friends: #{@dinner.custom_url}"
       redirect_to root_path
     else
+      flash.now[:error] = "Please complete all fields"
       render 'new'
     end
   end
 
+  def show
+    @dinner = Dinner.find_by_unique_url(params[:id])
```

Comment:
Careful here.  The semantics of `Model#find_by_<field>` are different than `Model#find`.  The former will return `nil` if no record is found whereas the latter will raise a `RecordNotFound` error (might have gotten the error class name wrong).